### DISCUSSION TITLE
Sometimes optional fields

### DISCUSSION THREAD

**Discussion by chris-hud** [spatie/laravel-data#506] [Q&A]
Is there an accepted practice to deal with sometimes optional fields? 
Optional fields are defined [here](https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/optional-properties) 

However, I often will have fields that are required on the model, but then create a method to update only a piece of the model, often only sending the relevant data over the wire, not the whole model. Essentially I need the fields to be optional for some requests, but not for others. 

```
order model
seller_name, required
buyer_name, required
seller_address, required
buyer_address, required
delivered, nullable

createOrder(Request $request) {
    Validator::make($request->all(), [
        'seller_name' => 'required|max:255',
        'buyer_name' => 'required|max:255',
        'seller_address' => 'required|max:255',
        'buyer_address' => 'required|max:255',
    ])->validate();
...
}

updateOrderDeliveryStatus(Request $request) {
  Validator::make($request->all(), [
        'order_id' => 'required|Integer',
        'delivered' => 'required|Boolean',
    ])->validate();
...
}

updateSellerName(Request $request) {
  Validator::make($request->all(), [
        'order_id' => 'required|Integer',
        'seller_name' => 'required|max:255',
    ])->validate();
...
}
```

**Comment by TheFerrettDeveloper** [comment]
Bumping because I too would like to know.

**Comment by hettiger** [comment]
I'm also interested in this. Bump

**Comment by backstageel** [comment]
I am also looking for this
